src/lib/operators/aggregate_hash.cpp

```diff
@@ -966,45 +959,29 @@ std::shared_ptr<SegmentVisitorContext> AggregateHash::_create_aggregate_context(
     using ColumnDataType = typename decltype(type)::type;
     switch (function) {
```

Comment:
Is there something on the C++ horizon that fixes such switches with template values?

Reply:
Wouldn't hold my breath for it given that cases are just glorified labels and have fallthrough semantics. That would defeat many of constexpr's advantages and together with legally ill-formed code, it would cause humongous headaches for compiler and application devs alike.